tests/integ/test_multithreading.py

```diff
@@ -256,30 +277,28 @@ def put_and_get_file(upload_file_path, download_dir):
 def test_concurrent_add_packages(threadsafe_session):
     # this is a list of packages available in snowflake anaconda. If this
     # test fails due to packages not being available, please update the list
+    existing_packages = threadsafe_session.get_packages()
     package_list = {
-        "graphviz",
+        "cloudpickle",
         "numpy",
         "pandas",
         "scipy",
         "scikit-learn",
-        "matplotlib",
+        "pyyaml",
     }
 
     try:
         with ThreadPoolExecutor(max_workers=10) as executor:
-            futures = [
+            for package in package_list:
                 executor.submit(threadsafe_session.add_packages, package)
-                for package in package_list
-            ]
 
-            for future in as_completed(futures):
-                future.result()
```

Comment:
we don't need to wait for the task to finish here?

Reply:
I moved `.get_packages()` outside of `with ThreadPoolExecutor()...` context. I learned from you that we come out after finishing all tasks. So made the test cleaner

Reply:
aha I see. it's a shame I forgot this already :P